storage-api/src/main/java/com/linagora/calendar/storage/MemoryResourceDAO.java

```diff
@@ -0,0 +1,130 @@
+import com.linagora.calendar.storage.model.Resource;
+import com.linagora.calendar.storage.model.ResourceId;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class MemoryResourceDAO implements ResourceDAO {
+    private final Map<ResourceId, Resource> store = new ConcurrentHashMap<>();
+
+    @Override
+    public Mono<ResourceId> insert(ResourceInsertRequest request) {
+        return Mono.fromCallable(() -> {
+            ResourceId newId = new ResourceId(UUID.randomUUID().toString());
+            Resource resource = new Resource(
+                newId,
+                request.administrators(),
+                request.creator(),
+                request.deleted(),
+                request.description(),
+                request.domain(),
+                request.icon(),
+                request.name(),
+                request.creation(),
+                request.updated(),
+                request.type()
+            );
+            store.put(newId, resource);
+            return newId;
+        });
+    }
+
+    @Override
+    public Flux<Resource> findAll() {
+        return Flux.fromIterable(store.values());
+    }
+
+    @Override
+    public Mono<Resource> findById(ResourceId id) {
+        return Mono.justOrEmpty(store.get(id));
+    }
+
+    @Override
+    public Flux<Resource> findByDomain(OpenPaaSId domainId) {
+        return Flux.fromStream(store.values().stream()
+            .filter(r -> r.domain().equals(domainId)));
+    }
+
+    @Override
+    public Mono<Resource> update(ResourceId id, ResourceUpdateRequest request) {
+        return Mono.just(Optional.ofNullable(store.get(id)).map(resource -> {
+            Resource updated = new Resource(
+                resource.id(),
+                request.administrators().orElse(resource.administrators()),
+                resource.creator(),
+                resource.deleted(),
+                request.description().orElse(resource.description()),
+                resource.domain(),
+                request.icon().orElse(resource.icon()),
+                request.name().orElse(resource.name()),
+                resource.creation(),
+                Instant.now(),
+                resource.type()
+            );
+            store.put(id, updated);
+            return updated;
+        }).orElseThrow(() -> new ResourceNotFoundException(id)));
+    }
+
+    @Override
+    public Mono<Void> softDelete(ResourceId id) {
+        return Mono.fromRunnable(() -> Optional.ofNullable(store.get(id)).map(resource -> {
+            Resource deleted = new Resource(
+                resource.id(),
+                resource.administrators(),
+                resource.creator(),
+                true,
+                resource.description(),
+                resource.domain(),
+                resource.icon(),
+                resource.name(),
+                resource.creation(),
+                Instant.now(),
+                resource.type()
+            );
```

Comment:
I would love us to have this POJO manipulation in POJOs in order to keep the business layers clean.

Suggestion:


```suggestion
            Resource deleted = resource.markedAsDeleted();
```

With `Resource::markedAsDeleted` containing the aforementioned code.